### DISCUSSION TITLE
Container is 970GB big

### DISCUSSION THREAD

**Discussion by timkolloch** [qdm12/ddns-updater#739] [General]
When I tried to install an apt package on my debian server I got the response that I did not have enough storage to download the package.
After some fiddeling around I saw that my ddns-updater container was 970GB leaving me with literally no space left on my hard drive.

The container probably ran for almost a year with our being removed but 970GB in logs or anything still seems way too much.

Did someone else make a similiar observation?

Sadly I needed to remove the  whole container because I really couldn't do anything so I cannot look into it anymore.

## Edit

I may have found a reason for why it got so big. I forgot that I made some changes to rule out a problem which left the container running (for a few day at most) on a port which was already in use this I always got an error message stating that. The container did not run for a year or so. However below is the log of not even one second (I needed to remove a lot because the body was too long). 

I do not know if a few days doing this might lead to 970GB in container size but it definetly played a role.

I guess one can argue "shit in shit out" but I think this gets logged way too often so one might want to change that.

```
2024-06-06T18:13:01Z INFO [http server] listening on :8000
2024-06-06T18:13:01Z ERROR [http server] listen tcp :8000: bind: address already in use
2024-06-06T18:13:01Z INFO [http server] restarting
2024-06-06T18:13:01Z INFO [http server] listening on :8000
2024-06-06T18:13:01Z ERROR [http server] listen tcp :8000: bind: address already in use...
2024-06-06T18:13:01Z INFO [http server] listening on :8000
2024-06-06T18:13:01Z ERROR [http server] listen tcp :8000: bind: address already in use
2024-06-06T18:13:01Z INFO [http server] restarting
2024-06-06T18:13:01Z INFO [http server] listening on :8000
```

**Comment by qdm12** [comment]
Similar to #720 - this is solved from commit 130ab008b5ec424665a6ee2557222ae8d4436184 - when one of the long running services such as the web server is not working, it crashes the entire container, instead of trying to restart internally.
On top of this, there is a new variable `SERVER_ENABLED=yes` you can set if you don't want the web ui, although this is more for the standalone binary instead of the container. Note however the healthcheck server (port 9999) is still mandatory for container setups, and cannot be disabled.

**Reply by timkolloch** [reply]
That ist perfect, thank you!
